backend/envconfig/envconfig.go

```diff
@@ -0,0 +1,79 @@
+package envconfig
+
+// TODO(issue#649): move this package into app framework
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strconv"
+
+	"github.com/short-d/app/fw"
+)
+
+// EnvConfig parses configuration from environmental variables.
+type EnvConfig struct {
+	environment fw.Environment
+}
+
+// ParseConfigFromEnv retrieves configurations from environmental variables and
+// parse them into the given struct.
+func (e EnvConfig) ParseConfigFromEnv(config interface{}) error {
+	configVal := reflect.ValueOf(config)
+	if configVal.Kind() != reflect.Ptr || configVal.IsNil() {
+		return errors.New("config must be a pointer")
+	}
+
+	elem := configVal.Elem()
+	if elem.Kind() != reflect.Struct {
+		return errors.New("config must be a struct")
+	}
+
+	numFields := elem.NumField()
+	configType := elem.Type()
+
+	for idx := 0; idx < numFields; idx++ {
+		field := configType.Field(idx)
+		envName, ok := field.Tag.Lookup("env")
+		if !ok {
+			continue
+		}
+		defaultVal := field.Tag.Get("default")
+		envVal := e.environment.GetEnv(envName, defaultVal)
+		err := setFieldValue(field, elem.Field(idx), envVal)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func setFieldValue(field reflect.StructField, fieldValue reflect.Value, newValue string) error {
+	kind := field.Type.Kind()
+	switch kind {
+	case reflect.String:
+		fieldValue.SetString(newValue)
+		return nil
+	case reflect.Int:
+		num, err := strconv.Atoi(newValue)
+		if err != nil {
+			return err
+		}
+		fieldValue.SetInt(int64(num))
+		return nil
+	case reflect.Bool:
+		boolean, err := strconv.ParseBool(newValue)
+		if err != nil {
+			return err
+		}
+		fieldValue.SetBool(boolean)
+		return nil
+	default:
+		return fmt.Errorf("unexpected field type: %s", kind)
```

Comment:
What about other types such as uint8, uint16, etc.? I think that there is no field other than string and int in ServiceConfig struct, is that the reason?

Reply:
@arberiii We’ll support more and more as needed! It doesn’t make sense for us to add everything at the moment ( we also have other priorities )